### PR TITLE
Add new gard clip [old gradient clip not support in dy graph]

### PR DESCRIPTION
**Pull request by phlrain**
old gradient clip not support in dy graph

### CODE REVIEW

File: python/paddle/fluid/optimizer.py [hunk 1]
```diff
@@ -496,6 +497,10 @@ def minimize(self,
             startup_program=startup_program,
             parameter_list=parameter_list,
             no_grad_set=no_grad_set)
+
+        if grad_clip is not None and framework.in_dygraph_mode():
+            params_grads = grad_clip(params_grads)
```
Comment (JiabinYang) [inline python/paddle/fluid/optimizer.py:507]: Add Warning to avoid static graph user to have unexpected effect

File: python/paddle/fluid/clip.py [hunk 2]
```diff
@@ -364,3 +367,253 @@ def append_gradient_clip_ops(param_grads):
 ClipByValue = GradientClipByValue
 ClipByNorm = GradientClipByNorm
 ClipByGlobalNorm = GradientClipByGlobalNorm
+
+
```
Comment (JiabinYang) [inline python/paddle/fluid/clip.py]: Maybe add TODO to clarify we will replace original grad_clip with this GradClip

**Review by shanyi15 — DISMISSED** [review]
LG

File: python/paddle/fluid/tests/unittests/test_grad_clip_minimize.py [hunk 3]
```diff
@@ -0,0 +1,254 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (JiabinYang) [inline python/paddle/fluid/tests/unittests/test_grad_clip_minimize.py:1]: Better to rename this test with `Dygraph`

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by chengduoZH — DISMISSED** [review]

**Review by XiaoguangHu01 — DISMISSED** [review]

**Review by shanyi15 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]

**Pull request merged**